RD-1833 Update filters docs (#138)
cloudify-cosmo/cloudify-rest-docs: content/_searches.md

# Searches

The `/searches/<resource>` endpoint is used to get a filtered list of a resource (currently, deployments or blueprints) 
based on its labels and certain attributes.
Deployments can be filtered be the following attributes: `blueprint_id`, `created_by`, `site_name`, and `schedules`.
Blueprints can be filters by the attribute `created_by`. 

Filtering can be done by specifying a pre-created filter ID, or by providing a list of filter rules. 
A filter rule is a dictionary of the following form: 
```text
{
 "key": "<key>",
 "values": [<list of values>],
 "operator": "<LabelsOperator>" or "<AttrsOperator>",
 "type": "<FilterRuleType>"
}
```
`<LabelsOperator>` can be one of: "any_of", "not_any_of", "is_null" or "is_not_null".

`<AttrsOperator>` can be one of: "any_of", "not_any_of", "contains", "not_contains", "starts_with", "ends_with", "is_not_empty".

`<FilterRuleType>` can be one of: "label" or "attribute". If "label" is provided, then the operator must be a `<LabelsOperator>`, and if "attribute" is provided, then 
the operator must be an `<AttrsOperator>`. 

> E.g. filtering by the following filter rules, will return all items of a resource that their creator name starts with "alice" or "bob", 

Reviewer: ```suggestion
E.g. filtering by the following filter rules, will return all items of a resource whose creator name starts with "alice" or "bob", 
```

Author: Changed here and in _filters.md
